### PR TITLE
ImageApprovalStatus updated

### PR DESCRIPTION
**Pull request by milosst94**

### CODE REVIEW

File: src/images/images.controller.ts [hunk 1]
```diff
@@ -17,6 +19,12 @@ import { ImagesService } from './images.service';
 export class ImagesController {
   constructor(private readonly imagesService: ImagesService) {}
 
+  @Put()
```
Comment (Branimir) [inline src/images/images.controller.ts]: This should be a specific route

File: src/images/images.service.ts [hunk 2]
```diff
@@ -17,6 +17,15 @@ export class ImagesService {
     private readonly watermarksRepository: Repository<Watermark>,
   ) {}
 
+  async updateImageApprovalStatus(@Body() imagesData: { id: number; isApproved: boolean }[]){
```
Comment (Branimir) [inline src/images/images.service.ts]: This should be just a variable, not from body

File: src/images/images.controller.ts [hunk 3]
```diff
@@ -17,6 +19,12 @@ import { ImagesService } from './images.service';
...
+  async updateImageApprovalStatus(@Body() imagesData: { id: number; isApproved: boolean }[]) {
+    await this.imagesService.updateImageApprovalStatus(imagesData)
```
Comment (Branimir) [inline src/images/images.controller.ts]: What do we return?

**Review by Branimir — CHANGES_REQUESTED** [review]

File: src/images/images.service.ts [hunk 4]
```diff
@@ -17,6 +17,15 @@ export class ImagesService {
...
+    const arrOfPromises = []
+    imagesData.forEach(element=>{
+      arrOfPromises.push(this.imagesRepository.update( element.id, {isApproved: element.isApproved}))
+      })
+    await Promise.all(arrOfPromises);
```
Comment (stefanj122) [inline src/images/images.service.ts]: Here you can do try, catch on this promises.

File: src/images/images.controller.ts [hunk 5]
```diff
@@ -17,6 +20,13 @@ import { ImagesService } from './images.service';
 export class ImagesController {
   constructor(private readonly imagesService: ImagesService) {}
 
+  @Put('images/Approval')
```
Comment (Branimir) [inline src/images/images.controller.ts]: Why Capital A? :)

File: src/images/images.service.ts [hunk 6]
```diff
@@ -17,6 +21,25 @@ export class ImagesService {
...
+        this.imagesRepository.update(element.id, {
+          isApproved: element.isApproved,
+        }),
+      );
+    });
+    try {
+      await Promise.all(arrOfPromises);
+      return 'success';
+    } catch (error) {
+      return BadRequestException;
```
Comment (Branimir) [inline src/images/images.service.ts]: We need to "throw" exception - return will give 200 response code, which is not good for FE developers

**Review by Branimir — CHANGES_REQUESTED** [review]

**Review by Branimir — APPROVED** [review]

**Review by Branimir — APPROVED** [review]

**Pull request closed without merging**